Modules/Scripted/VPAWModel/VPAWModel.py

```diff
@@ -253,31 +273,50 @@ def updateGUIFromParameterNode(self, caller=None, event=None):
         self._updatingGUIFromParameterNode = True
 
         # Update node selectors and sliders
-        self.ui.inputSelector.setCurrentNode(
-            self._parameterNode.GetNodeReference("InputVolume")
-        )
-        self.ui.outputSelector.setCurrentNode(
-            self._parameterNode.GetNodeReference("OutputVolume")
-        )
-        self.ui.invertedOutputSelector.setCurrentNode(
-            self._parameterNode.GetNodeReference("OutputVolumeInverse")
+        self.ui.PediatricAirwayAtlasDirectory.currentPath = (
+            self._parameterNode.GetParameter("PediatricAirwayAtlasDirectory")
         )
-        self.ui.imageThresholdSliderWidget.value = float(
-            self._parameterNode.GetParameter("Threshold")
+        self.ui.VPAWRootDirectory.currentPath = self._parameterNode.GetParameter(
+            "VPAWRootDirectory"
         )
-        self.ui.invertOutputCheckBox.checked = (
-            self._parameterNode.GetParameter("Invert") == "true"
+        self.ui.VPAWModelsDirectory.currentPath = self._parameterNode.GetParameter(
+            "VPAWModelsDirectory"
         )
 
         # Update buttons states and tooltips
-        if self._parameterNode.GetNodeReference(
-            "InputVolume"
-        ) and self._parameterNode.GetNodeReference("OutputVolume"):
-            self.ui.applyButton.toolTip = "Compute output volume"
-            self.ui.applyButton.enabled = True
+        self.ui.PediatricAirwayAtlasDirectory.toolTip = "Root directory for source code"
+        if os.path.isdir(self.ui.PediatricAirwayAtlasDirectory.currentPath):
+            self.ui.installPediatricAirwayAtlasButton.toolTip = (
+                "Install Pediatric Airway Atlas and Dependencies"
+            )
+            self.ui.installPediatricAirwayAtlasButton.enabled = True
+        else:
+            self.ui.installPediatricAirwayAtlasButton.toolTip = (
+                "Install is disabled; first select a valid source code directory for"
+                + " the Pediatric Airway Atlas"
+            )
+            self.ui.installPediatricAirwayAtlasButton.enabled = False
+
+        self.ui.VPAWRootDirectory.toolTip = (
+            "Directory containing FilteredControlBlindingLogUniqueScanFiltered.xls,"
+            + " images/*, and landmarks/*."
+        )
+        self.ui.VPAWModelsDirectory.toolTip = (
+            "Directory containing file named like '116(158.10-38.AM.24.Mar).pth'"
+        )
+        if os.path.isdir(self.ui.VPAWRootDirectory.currentPath) and os.path.isdir(
+            self.ui.VPAWModelsDirectory.currentPath
+        ):
+            self.ui.runPediatricAirwayAtlasButton.toolTip = (
+                "Run Pediatric Airway Atlas (This may take many hours)"
+            )
+            self.ui.runPediatricAirwayAtlasButton.enabled = True
```

Comment:
The enabled state of this button seems to toggle on properly, but it does not toggle off properly.

If I fill in directories for both `VPAWRootDirectory` and `VPAWModelsDirectory` then the button is enabled as expected. However if I then delete one of them and set it to be blank, the button remains enabled.

I am not sure why this is happening, maybe something to do with the triggers for `updateGUIFromParameterNode` to get called... not sure what triggers it exactly

Reply:
Good catch.  I have a fix: changing the string in one of those widgets to blank doesn't signal `currentPathChanged`, but it does signal `validInputChanged`.  So, now we'll connect both signals to the button enabling/disabling code.